src/model/CoilCoolingWater.cpp

```diff
@@ -273,48 +273,68 @@ namespace model {
     }
 
     bool CoilCoolingWater_Impl::addToNode(Node& node) {
-      bool success(false);
+      bool success = false;
 
       auto t_containingHVACComponent = containingHVACComponent();
       auto t_airLoop = node.airLoopHVAC();
 
-      if (t_airLoop && t_containingHVACComponent && t_containingHVACComponent->optionalCast<CoilSystemCoolingWaterHeatExchangerAssisted>()) {
-        LOG(Warn, this->briefDescription()
-                    << " cannot be connected directly to an AirLoopHVAC when it's part of a parent CoilSystemCoolingWaterHeatExchangerAssisted. "
-                       "Please call CoilSystemCoolingWaterHeatExchangerAssisted::addToNode instead");
-
-        /*       } else if (t_airLoop && t_containingHVACComponent && t_containingHVACComponent->optionalCast<CoilSystemCoolingWater>()) {
-        LOG(Warn, this->briefDescription()
-                    << " cannot be connected directly to an AirLoopHVAC when it's part of a parent CoilSystemCoolingWater. "
-                       "Please call CoilSystemCoolingWater::addToNode instead"); */
+      // Reject adding coils parent of a Parent System
+      if (t_airLoop && t_containingHVACComponent) {
+        if (t_containingHVACComponent->optionalCast<CoilSystemCoolingWaterHeatExchangerAssisted>()) {
+          LOG(Warn, this->briefDescription()
+                      << " cannot be connected directly to an AirLoopHVAC when it's part of a parent CoilSystemCoolingWaterHeatExchangerAssisted. "
+                         "Please call CoilSystemCoolingWaterHeatExchangerAssisted::addToNode instead");
+          return false;
+        }
 
-      } else {
+        if (t_containingHVACComponent->optionalCast<CoilSystemCoolingWater>()
+            && t_containingHVACComponent->cast<CoilSystemCoolingWater>().coolingCoil().handle() == this->handle()) {
+          LOG(Warn, this->briefDescription()
+                      << " cannot be connected directly to an AirLoopHVAC when it's the primary cooling coil of a parent CoilSystemCoolingWater. "
+                         "Please call CoilSystemCoolingWater::addToNode instead");
+          return false;
+        }
+      }
```

Comment:
The primary cooling coil of the CoilSystemCoolingWater shouldn't be connected directly to an AirLoopHVAC, at least that's my interpretation of the E+ testfiles.